### DISCUSSION TITLE
How should we generate the SVG?

### DISCUSSION THREAD

**Discussion by jsonsivar** [fleekxyz/non-fungible-apps#68] [Architecture] [answered]
First, we need to confirm how Uniswap is doing it and probably follow the best practice there.

I found contracts in their repo that suggest they do it on chain so we can perhaps follow suit: https://github.com/Uniswap/v3-periphery/blob/main/contracts/libraries/NFTSVG.sol.

Anyone worked with SVG generation before @zoruka @EmperorOrokuSaki @csosamorales?

Also just confirming, the tool @harrisonhines pointed us to can help us here @rollsmorr1 or we have to build this from scratch?

**Comment by EmperorOrokuSaki** [comment]
I will be working on this today. Will post an update here by EOD.

**Comment by EmperorOrokuSaki** [comment, marked as answer]
https://github.com/fleekxyz/non-fungible-apps/pull/70
